exchanges/btcmarkets/btcmarkets_websocket.go

```diff
@@ -79,65 +131,30 @@ func (b *BTCMarkets) wsHandleData(respRaw []byte) error {
 			return err
 		}
 
-		p, err := currency.NewPairFromString(ob.Currency)
-		if err != nil {
-			return err
-		}
-
-		var bids, asks orderbook.Items
-		for x := range ob.Bids {
-			var price, amount float64
-			price, err = strconv.ParseFloat(ob.Bids[x][0].(string), 64)
-			if err != nil {
-				return err
-			}
-			amount, err = strconv.ParseFloat(ob.Bids[x][1].(string), 64)
-			if err != nil {
-				return err
-			}
-			bids = append(bids, orderbook.Item{
-				Amount:     amount,
-				Price:      price,
-				OrderCount: int64(ob.Bids[x][2].(float64)),
-			})
-		}
-		for x := range ob.Asks {
-			var price, amount float64
-			price, err = strconv.ParseFloat(ob.Asks[x][0].(string), 64)
-			if err != nil {
-				return err
-			}
-			amount, err = strconv.ParseFloat(ob.Asks[x][1].(string), 64)
-			if err != nil {
-				return err
-			}
-			asks = append(asks, orderbook.Item{
-				Amount:     amount,
-				Price:      price,
-				OrderCount: int64(ob.Asks[x][2].(float64)),
-			})
-		}
 		if ob.Snapshot {
-			bids.SortBids() // Alignment completely out, sort is needed.
 			err = b.Websocket.Orderbook.LoadSnapshot(&orderbook.Base{
-				Pair:            p,
-				Bids:            bids,
-				Asks:            asks,
+				Pair:            ob.Currency,
+				Bids:            orderbook.Items(ob.Bids),
+				Asks:            orderbook.Items(ob.Asks),
 				LastUpdated:     ob.Timestamp,
+				LastUpdateID:    ob.SnapshotID,
 				Asset:           asset.Spot,
 				Exchange:        b.Name,
 				VerifyOrderbook: b.CanVerifyOrderbook,
 			})
 		} else {
 			err = b.Websocket.Orderbook.Update(&buffer.Update{
-				UpdateTime: ob.Timestamp,
-				Asset:      asset.Spot,
-				Bids:       bids,
-				Asks:       asks,
-				Pair:       p,
+				UpdateTime:          ob.Timestamp,
+				UpdateID:            ob.SnapshotID,
+				Asset:               asset.Spot,
+				Bids:                orderbook.Items(ob.Bids),
+				Asks:                orderbook.Items(ob.Asks),
```

Comment:
This casting is not actually required, but I do like it being explicit. No change required